src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,24 +1611,1318 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+    {
+        switch (param)
+        {
+            case ACTION_START_SCENE_FENRIS:
+            {
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                {
+                    me->CastSpell(player, SPELL_FORCE_CAST_FENRIS_CAMERA, true);
+                    player->NearTeleportTo(NoEscapeStartPos, true);
+                }
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_RESET_SCENE_FENRIS:
+                    Reset();
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+};
+
+Position const CameraFrontyardPos = { 980.7f, 689.14f, 76.9f };
+
+enum FenrisKeepCamera
+{
+    SPELL_SUMMON_FENRIS_ACTORS                  = 84053,
+    SPELL_GENERAL_TRIGGER_84102                 = 84102,
+    SPELL_FORCE_SEAT_2                          = 84091,
+    SPELL_DESPAWN_ALL_SUMMONS_FENRIS            = 84066,
+
+    EVENT_MOVE_TO_START_POINT                   = 1,
+    EVENT_CHANGE_TO_SEAT_2                      = 2,
+    EVENT_TRIGGER_84102                         = 3,
+    EVENT_SCENE_FINISH_FENRIS                   = 4,
+
+    POINT_CAMERA_FRONTYARD                      = 1,
+
+    SEAT_FENRIS_CAMERA                          = 0,
+    SEAT_FENRIS_CAMERA_FORCE                    = 1
+};
+
+// 45003 - Fenris Keep Camera
+struct npc_silverpine_fenris_keep_camera : public ScriptedAI
+{
+    npc_silverpine_fenris_keep_camera(Creature* creature) : ScriptedAI(creature) { }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+        {
+            _playerGUID = player->GetGUID();
+
+            if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                player->EnterVehicle(me, SEAT_FENRIS_CAMERA);
+        }
+
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        switch (spellInfo->Id)
+        {
+            case SPELL_GENERAL_TRIGGER_84107:
+                _events.ScheduleEvent(EVENT_SCENE_FINISH_FENRIS, 2s + 500ms);
+                break;
+            default:
+                break;
+        }
+    }
+
+    void PassengerBoarded(Unit* passenger, int8 seatId, bool apply) override
+    {
+        if (apply)
+        {
+            if (seatId == SEAT_FENRIS_CAMERA)
+            {
+                _events.ScheduleEvent(EVENT_MOVE_TO_START_POINT, 10ms);
+
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
```

Comment:
is player == passenger here?

Reply:
yes, should I be using passenger here rather than player instead?

Reply:
Feedback applied.